### PR TITLE
Added application info for /payments and modifications

### PR DESCRIPTION
**Pull request by lancergr**
**Description**
Added application info for /payments and modifications

**Tested scenarios**
card authorisation and capture

### CODE REVIEW

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java [hunk 1]
```diff
@@ -280,15 +310,30 @@ private String getCountryCode(CartData cartData) {
...
     }
 
     public RefundRequest createRefundRequest(final String merchantAccount, final BigDecimal amount, final Currency currency, final String authReference, final String merchantReference) {
-        return new RefundRequest().fillAmount(String.valueOf(amount), currency.getCurrencyCode()).merchantAccount(merchantAccount).originalReference(authReference).reference(merchantReference);
+        RefundRequest request = new RefundRequest().fillAmount(String.valueOf(amount), currency.getCurrencyCode())
+                                                   .merchantAccount(merchantAccount)
+                                                   .originalReference(authReference)
+                                                   .reference(merchantReference);
+        ApplicationInfo applicationInfo = updateApplicationInfo(request.getApplicationInfo());
+        request.setApplicationInfo(applicationInfo);
```
Comment (rikterbeek) [inline adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: shall we merge the setApplicationInfo into the new RefundRequest and merge it so we don't have to store it in a variable ?

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java [hunk 2]
```diff
@@ -429,7 +474,7 @@ private Name getShopperNameFromAddress(AddressData addressData) {
 
     /**
      * Set the required fields for using the OpenInvoice API
-     *
+     * <p>
```
Comment (rikterbeek) [inline adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: do we need this ?

File: adyenv6core/resources/adyenv6core.build.number [hunk 3]
```diff
@@ -6,5 +6,5 @@ group.id=com.adyen.v6
 module.name=platform-module
 name=adyenv6core
 vendor=adyen
-version=3.4.0
-version.api=3.4.0
+version=4.0.0-rc
+version.api=4.0.0-rc
```
[the remark this answers is not in the document]
Comment (lancergr) [reply adyenv6core/resources/adyenv6core.build.number]: marking that this is non-yet-released version (release candidate)

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java (hunk 2 above)
Comment (lancergr) [reply adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: added by auto-formatter

File: adyenv6core/src/com/adyen/v6/constants/Adyenv6coreConstants.java [hunk 4]
```diff
@@ -29,6 +29,9 @@
  */
 public final class Adyenv6coreConstants extends GeneratedAdyenv6coreConstants {
 	public static final String EXTENSIONNAME = "adyenv6core";
+
+	public static final String PLUGIN_NAME = "adyen-hybris";
+	public static final String PLUGIN_VERSION = "4.0.0-rc";
```
[the remark this answers is not in the document]
Comment (lancergr) [reply adyenv6core/src/com/adyen/v6/constants/Adyenv6coreConstants.java]: sure!

File: adyenv6core/resources/adyenv6core.build.number (hunk 3 above)
[the remark this answers is not in the document]
Comment (rikterbeek) [reply adyenv6core/resources/adyenv6core.build.number]: ok clear.

File: adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java (hunk 1 above)
Comment (rikterbeek) [reply adyenv6core/src/com/adyen/v6/factory/AdyenRequestFactory.java]: ok let's keep it as it is.

**Pull request merged**